Update pgvector.mdx (for 0.8.0) (#30414)
supabase/supabase: apps/docs/content/guides/database/extensions/pgvector.mdx

})
```

## Specific usage cases

### Queries with filtering

> If you use IVFFlat or HNSW index and filter results from the query 

Reviewer: It seems that this sentence is not complete.

> If you use IVFFlat or HNSW index and filter results from the query 

Reviewer 2: ```suggestion
If you use an IVFFlat or HNSW index and naively filter the results based on the value of another column, you may get fewer rows returned than requested.
```

> E.g. 

Reviewer: ```suggestion
For example
```

```
SELECT * FROM items WHERE category_id = 123 ORDER BY embedding <-> '[3,1,2]' LIMIT 5;
```

> filters results of vector column embedding by another column ```category_id```.

Reviewer: ```suggestion
filters results of vector column `embedding` by another column ```category_id```.
```

> ```
> SELECT * FROM items WHERE category_id = 123 ORDER BY embedding <-> '[3,1,2]' LIMIT 5;
> ```
> filters results of vector column embedding by another column ```category_id```.

Reviewer 2: ````suggestion
For example, the following query may return fewer than 5 rows, even if 5 corresponding rows exist in the database. This is because the embedding index may not return 5 rows matching the filter.

```
SELECT * FROM items WHERE category_id = 123 ORDER BY embedding <-> '[3,1,2]' LIMIT 5;
```
````

> Then you can use [specific syntax for iterative search](https://github.com/pgvector/pgvector/?tab=readme-ov-file#iterative-index-scans) to get exact number of results as in the query.
> 

Reviewer 2: ```suggestion
To get the exact number of requested rows, use [iterative search](https://github.com/pgvector/pgvector/?tab=readme-ov-file#iterative-index-scans) to continue scanning the index until enough results are found.
```
